src/main/java/ratpack/zipkin/ServerTracingModule.java

```diff
@@ -50,11 +51,27 @@ protected void configure() {
         .to(DefaultServerTracingHandler.class)
         .in(Singleton.class);
 
-    bind(HttpClient.class).annotatedWith(Zipkin.class)
-        .to(ZipkinHttpClientImpl.class)
+    bind(ClientTracingInterceptor.class)
+        .to(DefaultClientTracingInterceptor.class)
         .in(Singleton.class);
 
-    bind(ZipkinHttpClientImpl.class);
+    Provider<ClientTracingInterceptor> clientTracingInterceptorProvider =
+        getProvider(ClientTracingInterceptor.class);
+
+    Provider<HttpClient> httpClientProvider = () ->
+        Exceptions.uncheck(() -> HttpClient.of(spec -> {
```

Comment:
As originally pointed out by @beckje01 `This isn't perfect, I think it needs to be additive right now meaning it can't take over the HttpClient directly yet without ratpack changes...`

I'm still unsure how to address this.  As far as I can tell there is no way to make this additive but if anyone has ideas I'm open to trying other approaches.

Reply:
I don't think we can do any better right now

Reply:
I think I'm missing why the previous pattern of instantiating a second client, and have it only inject when the @Zipkin annotation exists doesn't work?

Reply:
Personally, I think I like it additive better as it allows me greater control over what I choose to trace.  Additionally, if there was a bug in this library - theoretically I could choose to remove the annotation and use the default client with no changes.

Reply:
Thanks @jeff-blaisdell.  Maybe we should raise an issue against Ratpack to introduce an API which takes an existing HttpClient that we can extend with additional config.

Reply:
@jeff-blaisdell would it be enough to _not_ replace the default HttpClient instance and only add the annotated version like before?  This doesn't solve the additive config problem but it would allow you to opt out of using the instrumented client.

Reply:
I just wasn't understanding the problem.  To paraphrase:
The intention was it would be nice if somehow we could extend default client configuration some configs are additive between the 2 clients rather than independent.  Agreed, that would be nice.

My point above is tangential to the config issue in that my preference is to have 2 independent client that I has the consumer of the library can choose between; rather than opt'ing me fully into Zipkin's variant.

Reply:
Does the http client instance need to be provided directly by the module?  Why not just provide the interceptors via the module and ask the application to register them with their http client?

Reply:
Think I have solved both concerns.  First I removed unannotated binding for `HttpClient` and second I changed the instrumented HttpClient provider to use `copyWith` to make the instrumentation additive to the default client as apposed to creating a new client.  This should allow modules configured before this one to set `HttpClient` configuration and this module will inherit the config.

Reply:
@janderson007 that would work as well however right now users expect that they can just use `@Zipkin` annotation to get an instrumented client.  Removing this means services would have to do more work and we would have to document this well.  Also in practice it was hard enough to communicate that you needed `@Zipkin` so I suspect that requiring more steps would bring even more questions when things don't work as expected.

Reply:
@llinder another point again the `@Zipkin` annotation as implemented is that it makes it impossible to disable zipkin by selectively installing/choosing not to install the zipkin module. We recently wanted to disable zipkin and had to make a code change to comment out the annotation. 
This might be a side PR but it would be nice to provide a config to the module to enable/disable tracing without code changes.

Reply:
@drmaas thats a great suggestion.  Adding some configuration to disable client instrumentation should be an easy add.  To solve the use case of being able to remove the Zipkin Module without requiring code changes, I think we would have to deprecate the use of `@Zipkin` annotation and replace the default HttpClient binding with the instrumented one.  In the end of the day, if someone wants both an instrumented and un-instrumented instance they could bind a new un-instrumented instance with their own annotation.

For now I will leave this PR as is and write up a ticket for your configuration suggestion.  Once this is merged it should be fairly painless to add the config change and cut a new release.  For now I still have to get the last test to pass so I will keep fiddling with that.